pkg/inject/sidecar.go

```diff
@@ -375,5 +375,6 @@ func EqualSidecar(dep, oldDep *appsv1.Deployment) bool {
 	}
 	depContainer := dep.Spec.Template.Spec.Containers[depAgentIndex]
 	oldDepContainer := oldDep.Spec.Template.Spec.Containers[oldDepIndex]
+	// Only compare args, if args are equal, assuming sidecars are equal
```

Comment:
What's "compare args" here? Arguments for the containers? The `DeepEqual` is comparing the whole container, isn't it?

Reply:
Old comment. I forgot to remove it.

Reply:
As soon as it's removed, I'll approve this PR and it will be merged when the CI passes.

Reply:
Removed.